quarkus-extension/engine/deployment/src/main/java/org/camunda/bpm/quarkus/engine/extension/deployment/impl/CamundaEngineProcessor.java

```diff
@@ -16,19 +16,75 @@
  */
 package org.camunda.bpm.quarkus.engine.extension.deployment.impl;
 
+import static io.quarkus.deployment.annotations.ExecutionTime.RUNTIME_INIT;
+
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.arc.deployment.BeanContainerBuildItem;
+import io.quarkus.arc.deployment.UnremovableBeanBuildItem;
+import io.quarkus.arc.processor.DotNames;
+import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.annotations.Record;
-import io.quarkus.deployment.annotations.ExecutionTime;
 import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.ShutdownContextBuildItem;
+import org.camunda.bpm.engine.cdi.BusinessProcess;
+import org.camunda.bpm.engine.cdi.ProcessVariables;
+import org.camunda.bpm.engine.cdi.compat.CamundaTaskForm;
+import org.camunda.bpm.engine.cdi.compat.FoxTaskForm;
+import org.camunda.bpm.engine.cdi.impl.ProcessVariableLocalMap;
+import org.camunda.bpm.engine.cdi.impl.ProcessVariableMap;
+import org.camunda.bpm.engine.cdi.impl.context.DefaultContextAssociationManager;
+import org.camunda.bpm.engine.cdi.jsf.TaskForm;
+import org.camunda.bpm.quarkus.engine.extension.impl.CamundaEngineRecorder;
+import org.jboss.jandex.DotName;
 
 public class CamundaEngineProcessor {
 
   protected static final String FEATURE = "camunda-platform-engine";
 
   @BuildStep
-  @Record(ExecutionTime.STATIC_INIT)
   protected FeatureBuildItem feature() {
     return new FeatureBuildItem(FEATURE);
   }
 
+  @BuildStep
+  protected void unremovableBeans(BuildProducer<UnremovableBeanBuildItem> unremovableBeansProducer) {
+    unremovableBeansProducer.produce(
+        UnremovableBeanBuildItem.beanTypes(
+            DotName.createSimple(DefaultContextAssociationManager.class.getName() + "$RequestScopedAssociation")));
```

Comment:
This is not very robust, but I think we don't need to find a solution in the scope of this ticket already since it might require a small refactoring in the `engine-cdi` module. With CAM-13735 I created a ticket to tackle it.

Reply:
Unfortunately, I'm not sure what would be the best way to refactor the `engine-cdi` module to make this line more robust. I would need to invest some time to understand it better. Since you already created CAM-13735 for it, I would suggest that we tackle it with that ticket.